include/ignition/gui/qml/GzColor.qml

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls 2.1
+import QtQuick.Dialogs 1.0
+import QtQuick.Layouts 1.3
+
+
+Item {
+  id: gzColorRoot
+  
+  implicitWidth: 40
+  implicitHeight: 40
+
+  property double r: 255
+  property double g: 0
+  property double b: 0
+  property double a: 1.0
+
+  signal colorSet()
+
+  Button {
+    id: colorButton
+    Layout.leftMargin: 5
+    ToolTip.text: "Open color dialog"
+    ToolTip.visible: hovered
+    ToolTip.delay: Qt.styleHints.mousePressAndHoldInterval
+    background: Rectangle {
+      implicitWidth: 40
+      implicitHeight: 40
+      radius: 5
+      border.color: Qt.rgba(r / 255, g / 255, b / 255, a)
```

Comment:
Currently, if I choose white for a color the color button isn't clearly visible. Let's have the border color be black when the material theme is light and white when the theme is dark. You should be able to use something like this: https://github.com/gazebosim/gz-gui/blob/c7bfb8acf88ce6f0e46aa5bb4d214e7184e41aea/include/ignition/gui/qml/IgnSnackBar.qml#L136